### PR TITLE
[Object,ELFType] Rename TargetEndianness to Endianness

### PR DESCRIPTION
**Pull request by MaskRay**
`TargetEndianness` is a long name and using it is unwieldy.
"Target" in the name is confusing. Rename it to "Endianness".

I cannot find noticeable out-of-tree users of `TargetEndianness`, but
keep `TargetEndianness` to make this patch safer. `TargetEndianness`
will be removed by a subsequent change.

### CODE REVIEW

**Review by jh7370 — COMMENTED** [review]
I know the aim is to shorten the name, but if I'm not mistaken, the term should be "Endianness". I'm pretty sure I've never seen the term "Endian" used outside the context of "big-endian" or "little-endian".

Also, if you want to shorten the name, why create a separate alias, rather than just renaming the original term?

**Comment by MaskRay** [comment]
> I know the aim is to shorten the name, but if I'm not mistaken, the term should be "Endianness". I'm pretty sure I've never seen the term "Endian" used outside the context of "big-endian" or "little-endian".

Thanks for the comment. I see that "endian" is used as an adjective. However, it seems used widely, e.g. the system header file `endian.h`. There are other cases a phrase that is not noun is used as variable names, e.g. `old`/`previous`/`next`, or `Is64Bits`. Do I have any chance to get the shorter `Endian` instead of `Endianness`? :)

> Also, if you want to shorten the name, why create a separate alias, rather than just renaming the original term?

There are quite a few uses of `TargetEndianness`. While my long term goal is indeed to remove them, I plan to keep it here so that potential users will more time for migration.

**Comment by jh7370** [comment]
TL;DR: I don't think we should be using the term "endian" in this context, and I think existing usages of it as a variable name is a mistake/not-conformant with our coding standards. It also saves a whole 4 characters, which, frankly, you shouldn't be worrying about. I can get behind dropping the "Target" prefix, since it is redundant in the context. "Endianness" is the more correct term and is much more widely used in this kind of context than "endian".

I also think it would be better to just do a blanket rename rather than create an alias and have a migration period. Out-of-tree targets are very unlikely to switch until forced. In-tree code is entirely within our control and there's no benefit in staggering the rename locally.

---

I fell down a linguistic rabbit hole: I did some searching of various online dictionaries, including one computing-specific one, and the term "endian" either a) doesn't appear at all, b) appears but only in the form "big-endian" and "little-endian", or c) in one sole case, standalone, but only as a suffix (so not really standalone). Wikipedia's page on the topic uses the term "endianness" only. I was about to declare that "endian" isn't even really a word, and the usage within LLVM's codebase is a mistake. Then I stumbled upon the C++20 `std::endian` enum. So it looks like the committee has decreed that "endian" is a legitimate term (though note that the discussion https://groups.google.com/a/isocpp.org/g/std-proposals/c/Hx4ipachs1E/m/yCxriKIvEgAJ and linked proposal primarily used "endianness").

LLVM coding standard states that variables should be nouns. There's a clear precedent for ignoring this for boolean-like types, to allow these to be phrases (e.g. `Is64Bits`), so something like `IsLittleEndian` is fine, but that's not applicable here. The case of `next`/`previous` is a prior art thing due to traditional linked list-like structures. I'm not sure I'd personally ever use a member name `old` (I'd use a full name like `oldThing`). Either way, that case doesn't apply here either.

**Comment by MaskRay** [comment]
Thank you for the research. I've switched from "Endian" to "Endianness". While 4 character longer, this PR will still remove 6 characters (`Target`, which puzzled me "is there a host != target" difference)...

**Comment by arichardson** [comment]
I guess the slightly shorter version could be `ByteOrder` but endianess is probably better. I agree that the Target prefix does not make any sense since it's the endianess for a given elf type and is unrelated to host/target.

**Review by arichardson — APPROVED** [review]

**Review by jh7370 — APPROVED** [review]
LGTM, thanks!
